### PR TITLE
COMMERCE-1458 check if definition-option rel is already defined befor…

### PR DESCRIPTION
**Pull request by luca-pellizzon**
…e creating a new one

### CODE REVIEW

**Comment by luca-pellizzon** [comment]
ci:test

File: commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionOptionRelLocalServiceImpl.java [hunk 1]
```diff
@@ -71,6 +71,14 @@ public CPDefinitionOptionRel addCPDefinitionOptionRel(
 			ServiceContext serviceContext)
 		throws PortalException {
 
+		CPDefinitionOptionRel cpDefinitionOptionRel =
```
Comment (marco-leo) [inline commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionOptionRelLocalServiceImpl.java:74]: @luca-pellizzon we can't add 2 same options to a product?

Comment (luca-pellizzon) [reply commerce-product-service/src/main/java/com/liferay/commerce/product/service/impl/CPDefinitionOptionRelLocalServiceImpl.java:74]: you can add more specifications but only 1 option per type

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by marco-leo** [comment]
ci:test

**Comment by ethanbustad** [comment]
ci:test
Error is `npm ERR! write after end`, script retries but fails repeatedly. Not sure any of our code could cause this, so trying the tests again to see if it goes away.

**Comment by ethanbustad** [comment]
ci:test
Tests may pass now that https://github.com/liferay/com-liferay-commerce/pull/1436 is merged in.

**Comment by luca-pellizzon** [comment]
ci:test

**Comment by luca-pellizzon** [comment]
ci:test

**Review by ethanbustad — APPROVED** [review]

**Comment by marco-leo** [comment]
I can add the same option many times to a product

**Comment by luca-pellizzon** [comment]
@marco-leo you can't because of this index:
create unique index IX_E57A1C2A on CPDefinitionOptionRel (CPDefinitionId, CPOptionId);

**Pull request closed without merging**